oauth-client/src/main/java/io/strimzi/kafka/oauth/client/JaasClientOauthLoginCallbackHandler.java

```diff
@@ -99,9 +100,9 @@ public void configure(Map<String, ?> configs, String saslMechanism, List<AppConf
             usernameClaim = null;
         }
 
-        isJWT = !config.getValueAsBoolean(Config.OAUTH_TOKENS_NOT_JWT, false);
-        if (!isJWT && usernameClaim != null) {
-            throw new RuntimeException("Custom username claim (OAUTH_USERNAME_CLAIM) not available, when tokens are configured as opaque (OAUTH_TOKENS_NOT_JWT)");
+        isJwt = isAccessTokenJwt(config, log, null);
+        if (!isJwt && usernameClaim != null) {
+            throw new RuntimeException("Custom username claim (OAUTH_USERNAME_CLAIM) not available, when tokens are configured as opaque (OAUTH_ACCESS_TOKEN_IS_JWT=false)");
```

Comment:
Should we have a more specific exception type for this?

Reply:
We don't handle this exception anywhere. Any exception thrown here is caught by Kafka Broker and results in broker shutting down. 

I don't think there is some Kafka runtime exception that _should_ be used here? If that were the case we should replace with it all occurrences of RuntimeException inside configure() method.

Reply:
I'm just thinking that if your Kafka broker shuts down in these circumstances it's slightly nicer to see `OauthConfigurationException` in the logs than `RuntimeException`.

Reply:
That's very subjective. I personally don't see that as added value. But since it would require changing all RuntimeException throw statements in configure(), if we really want that, I prefer we make a new PR just for that and keep this one focused on configuration alignment. Maybe create a JIRA for the other one if you think we really should introduce a new exception.